### PR TITLE
feat: add member to default activity area on sub join

### PR DESCRIPTION
**Pull request by zesinho**
- Add a default activity area to subtypes
- adds member to the default activity area if added to the related sub
- contains migration on Subscri

TODO:
- [ ] regenerate translations files. What is the command for that?
- [x] add tests if feature request was understood correctly (see comment on issue #452 )

### CODE REVIEW

File: juntagrico/views_create_subscription.py [hunk 1]
```diff
@@ -202,9 +203,11 @@ def get_initial(self):
         return {'comment': getattr(self.cs_session.main_member, 'comment', '')}
 
     def get_context_data(self, **kwargs):
+        args = self.cs_session.to_dict()
+        args['activity_areas'] = ActivityArea.objects.filter(auto_add_new_members=True)
```
Comment (motu55) [inline juntagrico/views_create_subscription.py:207]: Put this in the corresponding DAO

File: juntagrico/views_subscription.py [hunk 2]
```diff
@@ -306,11 +307,23 @@ def activate_subscription(request, subscription_id):
...
     try:
         subscription.activate(change_date)
+        add_subscription_member_to_activity_area(subscription)
     except ValidationError as e:
         return error_page(request, e.message)
     return return_to_previous_location(request)
 
 
+def add_subscription_member_to_activity_area(subscription):
+    activity_area = ActivityArea.objects.filter(auto_add_new_members=True)
```
Comment (motu55) [inline juntagrico/views_subscription.py:317]: Reuse DAO

File: juntagrico/views_subscription.py [hunk 3]
```diff
@@ -306,11 +307,23 @@ def activate_subscription(request, subscription_id):
...
+    member_ids = SubscriptionMembership.objects.filter(subscription=subscription).values_list('member')
```
Comment (motu55) [inline juntagrico/views_subscription.py:318]: There is a property for the recepients in the subscription, which should be used here

File: juntagrico/views_subscription.py [hunk 4]
```diff
@@ -306,11 +307,23 @@ def activate_subscription(request, subscription_id):
...
+    members = Member.objects.filter(pk__in=member_ids)
+
+    for member in members:
+        for area in activity_area:
+            if member not in area.members.all():
+                area.members.add(member)
+
```
Comment (motu55) [inline juntagrico/views_subscription.py:325]: [area.members.add(*members) for area in activity_area]
though not sure if *members works eventually you need to convert members to a list

File: juntagrico/templates/createsubscription/summary.html [hunk 5]
```diff
@@ -89,6 +89,18 @@ <h4>{{ v_co_member_pl }}</h4>
...
+        {% if activity_areas %}
+            <h4>{% trans "Tätigkeitsbereiche" %}</h4>
+            <p>{% blocktrans %}Du wird automatisch in folgende Tätigkeitsbereiche eingetragen.{% endblocktrans %}</p>
+            <ul>
+                {% for area in activity_areas %}
+                    <li><strong>{{ area.name }}</strong> {{ area.description |safe }}</li>
+                {% endfor %}
+            </ul>
+
+            <p>{% blocktrans %}Du kannst das später in den Einstellungen ändern.{% endblocktrans %}</p>
```
Comment (motu55) [inline juntagrico/templates/createsubscription/summary.html:101]: du kannsts das jederzeit unter dem Menupunkt Tätigkeitsbereiche ändern

File: juntagrico/entity/jobs.py [hunk 6]
```diff
@@ -28,6 +28,9 @@ class ActivityArea(JuntagricoBaseModel):
     members = models.ManyToManyField(
         'Member', related_name='areas', blank=True, verbose_name=Config.vocabulary('member_pl'))
     sort_order = models.PositiveIntegerField(_('Reihenfolge'), default=0, blank=False, null=False)
+    auto_add_new_members = models.BooleanField(_('Standard Tätigkeitesbereich für neue Benutzer'), default=False,
+                                               help_text=_(
+                                                   'Neue Benutzer automatisch zu diesem Tätigkeitsbereich hinzufügen.'))
```
Comment (motu55) [inline juntagrico/entity/jobs.py:33]: Neue benutzer werden automatisch zu diesem tätigkereitsbereich hinzugefügt

File: juntagrico/views_create_subscription.py [hunk 7]
```diff
@@ -202,9 +203,11 @@ def get_initial(self):
         return {'comment': getattr(self.cs_session.main_member, 'comment', '')}
 
     def get_context_data(self, **kwargs):
+        args = self.cs_session.to_dict()
```
Comment (motu55) [inline juntagrico/views_create_subscription.py:206]: could we not put the areas in here 
https://github.com/juntagrico/juntagrico/blob/f12079c51572f9bfc55f653d2bd8543d0212e555/juntagrico/util/sessions.py#L42
?

**Review by motu55 — CHANGES_REQUESTED** [review]
also the gnereate testdata and test data advanced commands need to be adapted

File: juntagrico/views_create_subscription.py (hunk 1 above)
Comment (zesinho) [reply juntagrico/views_create_subscription.py]: done

File: juntagrico/views_subscription.py (hunk 2 above)
Comment (zesinho) [reply juntagrico/views_subscription.py]: done

File: juntagrico/views_subscription.py (hunk 3 above)
Comment (zesinho) [reply juntagrico/views_subscription.py]: done

File: juntagrico/views_subscription.py (hunk 4 above)
Comment (zesinho) [reply juntagrico/views_subscription.py:318]: done, works

File: juntagrico/entity/jobs.py (hunk 6 above)
Comment (zesinho) [reply juntagrico/entity/jobs.py]: done

File: juntagrico/views_create_subscription.py (hunk 7 above)
Comment (zesinho) [reply juntagrico/views_create_subscription.py:206]: You could, but right now the areas are only used in the summary and nowhere else. Storing it in the sessions is not really necessary.

But it's up to you. This is a quick change.

Comment (zesinho) [reply juntagrico/views_create_subscription.py:206]: You could, but right now the areas are only used in the summary and nowhere else. Storing it in the sessions is not really necessary.

But it's up to you. This is a quick change.

**Comment by zesinho** [comment]
requested changes done and updated management testdata commands.

**Pull request merged**